CONTRIBUTING.md

```diff
@@ -0,0 +1,481 @@
+
+It is possible that the file already exists so be mindful of that.
+
+For example, let's decompile the first function in `asm/electric.s`:
+
+```asm
+    thumb_func_start sub_80ADBEC
+sub_80ADBEC: @ 80ADBEC
+    push {r4,lr}
+    adds r4, r0, 0
+    ldr r0, _080ADC0C @ =gBattleAnimAttacker
+    ldrb r0, [r0]
+    bl GetBattlerSide
+    lsls r0, 24
+    cmp r0, 0
+    beq _080ADC14
+    ldr r0, _080ADC10 @ =gBattleAnimArgs
+    ldrh r1, [r4, 0x20]
+    ldrh r2, [r0]
+    subs r1, r2
+    strh r1, [r4, 0x20]
+    b _080ADC20
+    .align 2, 0
+_080ADC0C: .4byte gBattleAnimAttacker
+_080ADC10: .4byte gBattleAnimArgs
+_080ADC14:
+    ldr r1, _080ADC34 @ =gBattleAnimArgs
+    ldrh r0, [r1]
+    ldrh r2, [r4, 0x20]
+    adds r0, r2
+    strh r0, [r4, 0x20]
+    adds r0, r1, 0
+_080ADC20:
+    ldrh r0, [r0, 0x2]
+    ldrh r1, [r4, 0x22]
+    adds r0, r1
+    strh r0, [r4, 0x22]
+    ldr r0, _080ADC38 @ =sub_80ADC3C
+    str r0, [r4, 0x1C]
+    pop {r4}
+    pop {r0}
+    bx r0
+    .align 2, 0
+_080ADC34: .4byte gBattleAnimArgs
+_080ADC38: .4byte sub_80ADC3C
+thumb_func_end sub_80ADBEC
+```
+
+#### Creating the file and linking it in ROM
+
+Create a file with the same name in `src/`. The first line in this new file
+should be `#include "global.h"`. `global.h` contains all of the type 
+definitions and many gba defines.
+
+Search for the file in the linker script `ld_script.txt`:
+
+```
+    asm/fire.o(.text);
+    asm/electric.o(.text);
+    asm/ice.o(.text);
+```
+
+and add the new C file BEFORE the assmebly file, since we are starting to
+decompile this file in order from the top.
+
+```
+    asm/fire.o(.text);
+    src/electric.o(.text);
+    asm/electric.o(.text);
+    asm/ice.o(.text);
+```
+
+#### Identifying the function signature
+
+We begin by trying to identify the function signature (the return type, and
+any arguments).
+
+To determine the return type of the function, we look at the bottom where the 
+function ends. We see `bx r0` which quickly tells us that our function is 
+`void`.
+
+Other branches you might see are `bx r1` or `bx lr`. Seeing these means you'll 
+need to infer the return type from the way it is called, and the values in the
+registers at the end, but it will never.
+
+To determine the arguments this function takes, we look at its beginning. The
+registers `r0`, `r1`, `r2`, and `r3` are used for arguments and return values.
+
+The function only preserves the value in `r0`, which tells us that it only 
+takes a single argument. Since it is not clearing any bits, we also know that
+it takes up a fill register, which is 4 bytes. Looking ahead in the function, 
+we see that it is accessed by specific offsets `ldrh r1, [r4, 0x20]`, which
+tells us that this argument is probably a struct pointer. More context confirms
+that is it a pointer to a Sprite.
+
+Our C file now looks like this:
+
+```C
+#include "global.h"
+#include "sprite.h"
+
+void sub_80ADBEC(struct Sprite *sprite)
```

Comment:
Italicization starts being used here for some reason and ends at line 244.

Reply:
That's just GitHub being a dum-dum